### ISSUE TITLE
MIDI File parse bug.

### ISSUE THREAD

**Issue by SinanKarasu** [orchetect/MIDIKit#177]
### Please Confirm

- [X] I have reviewed the **[MIDIKit Documentation](https://orchetect.github.io/MIDIKit)** which contains descriptive guides and extensive API reference
- [X] I have searched **[Issues](https://github.com/orchetect/MIDIKit/issues?q=is%3Aissue)** and **[Discussions](https://github.com/orchetect/MIDIKit/discussions)** to see if the same question has already been asked

### macOS Version(s) Used to Build

macOS 13 Ventura

### Xcode Version(s)

Xcode 14

### Description

The following file does not parse. The same file parses in two other MIDI Parsers, Logic Pro, and plays in AVAudioSequencer.
[Anonim - Uzun Ince Bir Yoldayim.mid.zip](https://github.com/orchetect/MIDIKit/files/10975442/Anonim.-.Uzun.Ince.Bir.Yoldayim.mid.zip)

**Comment by orchetect** [comment]
Thanks for the bug report. I've looked into this and the file is being successfully parsed, but there are two trailing bytes at the end of the file: 0x0D 0x0A. Because these bytes are unexpected, the file is considered "malformed".

Thus far I cannot find any reason for them being there. There is nothing in the SMF spec that mentions what they would be for. It's possible to ignore them but I'd rather get to the bottom of why they might be there in the first place.

**Comment by SinanKarasu** [comment]
I have developed parsers using Antlr, and this was always a problem between Windows and Unix. Here is one from IBM.
https://www.ibm.com/support/pages/additional-carriage-return-and-line-feed-added-during-sterling-connectdirect-file-transfer

**Comment by orchetect** [comment]
Yeah it's a Windows CRLF byte pair. Really odd. But if someone attempted editing the fine in a text editor then maybe that explains it. Some of the MIDI files out in the wild have survived decades and who knows what happened along the way.

On a related note: I sourced a zip file of about 30,000 crowd-sourced MIDI files from the web and ran all of them through a test harness that had `MIDIFile` parse them to catch any misfits. It returned a number of results so I'm going to go through those soon and see if the parser can be made more resilient to malformed or unexpected files.
